modules/drivers/mongo/test/metabase/driver/mongo/query_processor_test.clj

```diff
@@ -87,16 +88,22 @@
                         :query       [{"$match"
                                        {:$expr
                                         {"$eq"
-                                         [{:$let {:vars {:parts {:$dateToParts {:date "$datetime"}}}
-                                                  :in   {:$dateFromParts {:year "$$parts.year", :month "$$parts.month"}}}}
+                                         [{:$let {:vars {:parts {:$dateToParts {:date "$datetime"
+                                                                                :timezone (qp.timezone/results-timezone-id :mongo mt/db)}}}
```

Comment:
I know there are currently no drivers that derive from `:mongo` (at least that we know of) but you should still be explicitly passing around the `driver` parameter instead of assuming it's `:mongo`. If anything it will at least make it easier to mock stuff for test purposes.

```suggestion
                                                                                :timezone (qp.timezone/results-timezone-id driver mt/db)}}}
```

#### Here's a longer explanation of why this is important

Multimethods that dispatch on `driver` must explicitly pass the `driver` parameter along to other multimethods, instead of ignoring it and assuming the driver for which it is defined. Why? Because that method may in turn call other methods, and we might not be using the correct overrides anymore if `driver` was something other than what we assumed. 

The `:postgres` driver is a good example. `:redshift` derives from `:postgres`, and only overrides a handful of methods; it is a very bad assumption to assume that all Postgres methods are being called for `:postgres` and not for `:redshift`. By hardcoding the driver to `:postgres` we'll end up using `:postgres` methods for everything even if a `:redshift` specific override exists.

Here's an example:

```clj
;;; WRONG
(defmethod sql.qp/date [:postgres :week]
  [_ _ expr]
  (sql.qp/adjust-start-of-week :postgres (partial date-trunc :week) expr))
```

`sql.qp/adjust-start-of-week` in turn calls the multimethod `sql.qp/add-interval-honeysql-form`. This works perfectly fine for `:postgres`, but perhaps Redshift has its own implementation of `sql.qp/add-interval-honeysql-form`. We wouldn't be using because we hardcoded `:postgres` here.

The correct way to do this is:

```clj
;;; GOOD
(defmethod sql.qp/date [:postgres :week]
  [driver _unit expr]
  (sql.qp/adjust-start-of-week driver (partial date-trunc :week) expr))
```

Now `driver` is passed along and any `:redshift` specific overrides can be used.

Reply:
I'm aware of the hierarchy thing, I just didn't know that `(mt/test-driver :mongo ...)` would allow drivers other than `:mongo` and that `driver` is a valid reference at this point.

Reply:
@camsaul The suggested change doesn't seem to work (`driver` cannot be resolved). Are you suggesting changing the `(mt/test-driver :mongo ...)` forms everywhere or some other change?